openmarket-server/src/main/java/co/unicauca/openmarket/server/infra/OpenMarketHandler.java

```diff
@@ -0,0 +1,267 @@
+import co.unicauca.openmarket.commons.domain.Product;
+import co.unicauca.openmarket.commons.domain.ShoppingCart;
+import co.unicauca.openmarket.commons.domain.Category;
+import co.unicauca.openmarket.commons.domain.StateProduct;
+import co.unicauca.openmarket.commons.domain.User;
+import co.unicauca.openmarket.commons.infra.JsonError;
+import co.unicauca.openmarket.commons.infra.Protocol;
+import co.unicauca.openmarket.server.access.IProductRepository;
+import co.unicauca.openmarket.server.domain.service.OpenMarketFacade;
+import co.unicauca.strategyserver.infra.ServerHandler;
+
+import com.google.gson.Gson;
+
+import java.text.SimpleDateFormat;
+import java.text.DateFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+/**
+ * @author Braian Rey / Arturo Restrepo Ruiz
+ */
+public class OpenMarketHandler {
+    private Map<String, Consumer<Protocol>> actionMap;
+    private OpenMarketFacade facade;
+    User requester; 
+
+    public OpenMarketHandler(OpenMarketFacade facade, String requestJson) {
+        this.facade = facade;
+        
+        actionMap = new HashMap<>();
+        String UserJson = protocolRequest.getParameters().get(0).getValue();
+        User requester = gson.fromJson(userJson, User.class);
+        facade.setRequester(requester);
+        Gson gson = new Gson();
+        Protocol protocolRequest = gson.fromJson(requestJson, Protocol.class);
+        String reply = "";
+
+        // Crear un HashMap para mapear las acciones según el recurso
+        actionMap.put("findAvailableProducts", this::processFindAvailableProducts);
+        actionMap.put("findProductByNameAndDescription", this::processFindProductByNameAndDescription);
+        actionMap.put("buyProduct", this::processBuyProduct);
+        actionMap.put("addProductToShoppingCart", this::processAddProductToShoppingCart);
+        actionMap.put("getShoppingCart", this::processGetShoppingCart);
+        actionMap.put("buyShoppingCart", this::processBuyShoppingCart);
+        actionMap.put("deleteShoppingCart", this::processDeleteShoppingCart);
+        actionMap.put("saveProduct", this::processSaveProduct);
+        actionMap.put("updateProduct", this::processUpdateProduct);
+        actionMap.put("getOwnProductst", this::processGetOwnProducts);
+        actionMap.put("getOrders", this::processGetOrders);
+        actionMap.put("confirmOrder", this::processConfirmOrder);
+        actionMap.put("qualificateOrder", this::processQualificateOrder);
+        actionMap.put("registerDelivery", this::processRegisterDelivery);
+        // Obtener la clave para el HashMap
+        String key = protocolRequest.getResource() + "." + protocolRequest.getAction();
+        // Obtener el Consumer correspondiente y ejecutarlo
+        Consumer<Protocol> action = actionMap.get(key);
+        if (action != null) {
+            action.accept(protocolRequest);
```

Comment:
in en el metodod que vas a crear que es de proccesRequest queda String reply  = action.apply(protocolRequest); esi que va a ser la string que nos va a devolver cada metodo

Reply:
y pues ese metodod retornaria la string the respuesta

Reply:
En efecto, ya no tendrias que poner en cada metodo reply =